ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -1538,18 +1551,14 @@ public static void mvFileToFinalPath(Path specPath, Configuration hconf,
 
         // move to the file destination
         Utilities.FILE_OP_LOGGER.trace("Moving tmp dir: {} to: {}", tmpPath, specPath);
-
-        perfLogger.PerfLogBegin("FileSinkOperator", "RenameOrMoveFiles");
-        if (isBlobStorage) {
-          // HIVE-17113 - avoid copying files that may have been written to the temp dir by runaway tasks,
-          // by moving just the files we've tracked from removeTempOrDuplicateFiles().
-          Utilities.moveSpecifiedFiles(fs, tmpPath, specPath, filesKept);
+        if(shouldAvoidRename(conf, hconf)){
+          LOG.debug("Skipping rename/move files. Files to be kept are: " + filesKept.toString());
+          conf.getFilesToFetch().addAll(filesKept);
```

Comment:
We already have statusList. I am wondering using that instead of filesKept will let us avoid doing listStatus() call later.

Reply:
Good idea. Let me see if this could be done.